lib/errors.js

```diff
@@ -185,4 +185,29 @@ errors.GatewayTimeoutError = function(message) {
 };
 util.inherits(errors.GatewayTimeoutError, errors.SmartcarError);
 
+/**
+ * Enhanced errors from API v2.0
+ *
+ * @param {Object|String} error - response body from a v2.0 request
+ */
+class SmartcarErrorV2 extends errors.SmartcarError {
+  constructor(error) {
+    super();
+    if (error instanceof String) {
+      this.message = error;
+      return;
+    }
+    // standard v2 error attributes
```

Comment:
```suggestion
   super(error.description || error);
   this.name = 'SmartcarErrorV2';
   
    if (error instanceof String) {
      return;
    }
    
    // standard v2 error attributes
```

couple of things here: 
1. use the super-constructor to set `this.message` 
2. set `this.message` in the common case too (i.e. `error` is an object)
3. set the name of the class to `SmartcarErrorV2` (it's currently being set to `smartcar_error` by `super`)